4/로또의최고순위와최저순위-2.js

```diff
@@ -0,0 +1,20 @@
+function solution(lottos, win_nums) {
+  const rank = [];
+  let minRank = 0;
+  let count = 0;
+
+  for (let i in win_nums) {
+    if (lottos.includes(win_nums[i])) {
+      minRank += 1;
+    }
+
+    if (lottos[i] === 0) count += 1;
+  }
+
+  const maxRank = count + minRank;
+
+  rank.push(7 - (maxRank < 1 ? 1 : maxRank));
+  rank.push(7 - minRank >= 6 ? 6 : 7 - minRank);
+
```

Comment:
```suggestion
  const rank = [];
  let minRank = 0;
  let count = 0;

  for (let i in win_nums) {
    if (lottos.includes(win_nums[i])) {
      minRank += 1;
    } else if (lottos[i] === 0) {
      count += 1;
    }
  }

  const maxRank = count + minRank;

  rank.push(7 - (maxRank < 1 ? 1 : maxRank), 7 - minRank >= 6 ? 6 : 7 - minRank);

```
1. else 붙여주세요! 
2. NOT 연산자의 경우 if문의 조건식이 0인 경우도 NOT의 경우로 포함됩니다. 
3. 그리고 array.filter() 메서드를 활용하시면 for문을 사용하지 않아도 구현할 수 있어요. 
    (반복문 보다는 map,filter,reduce 메서드 사용을 지향합니다.)
```suggestion
  const rank = [];
  let minRank = 0;
  let count = 0;

  const count = lottos.filter( a => a !0).length;
  const  minRank = lottos.filter( b => win_nums.includes(b) > 0 ).length;

  const maxRank = count + minRank;

  rank.push(7 - (maxRank < 1 ? 1 : maxRank), 7 - minRank >= 6 ? 6 : 7 - minRank);

```
그리고 rank.push 값들이 가독성이 떨어지는 것 같아서 따로 순위에 대한 배열을 선언하고 해당 인덱스 값을 push하는 방법도 생각해보시면 좋을 것 같네요:)